cetlvast/suites/docs/examples/example_05_array_memory_resource_array.cpp

```diff
@@ -0,0 +1,120 @@
+                  << std::endl;
+
+        for (std::size_t i = 0; i < item_cout; ++i)
+        {
+            a.data.push_back(i);
+        }
+
+        std::cout << "AFTER  -> data size = " << a.data.size() << ", data capacity : " << a.data.capacity()
+                  << std::endl;
+        //![example_a]
+    }
+    {
+        //![example_b]
+        // BUT WAIT! THERE'S MORE! The UnsynchronizedArrayMemoryResource both slices and dices! That is, you can provide
+        // an upstream allocator to turn this into a "small buffer optimization" resource where the internal allocation
+        // is the small buffer and the upstream allocator becomes the larger allocator.
+        cetl::pmr::UnsynchronizedArrayMemoryResource<cetl::pmr::memory_resource>
+            bResource{small_message_buffer_, SmallMessageSizeBytes, cetl::pmr::new_delete_resource()};
+        cetl::pmr::polymorphic_allocator<std::uint64_t> bAlloc{&bResource};
+        Message                                         b{bAlloc};
+
+        // This time we won't reserve which should cause vector to do multiple allocations. We'll also insert
+        // a bunch more items then there space in the small message buffer.
+        std::cout << "BEFORE -> data size = " << b.data.size() << ", data capacity : " << b.data.capacity()
+                  << std::endl;
+
+        const std::size_t item_cout = (SmallMessageSizeBytes / sizeof(decltype(bAlloc)::value_type)) * 100;
+        for (std::size_t i = 0; i < item_cout; ++i)
+        {
+            b.data.push_back(i);
+        }
+
+        std::cout << "AFTER  -> data size = " << b.data.size() << ", data capacity : " << b.data.capacity()
+                  << std::endl;
+        //![example_b]
+    }
+    {
+        //![example_c]
+        // One more example: by using another UnsynchronizedArrayMemoryResource as an upstream for another
+        // UnsynchronizedArrayMemoryResource with the same-sized buffer you can use vector push_back without reserve up
+        // to the size of these buffers.
+        static cetl::byte upstream_buffer[SmallMessageSizeBytes];
+        cetl::pmr::UnsynchronizedArrayMemoryResource<cetl::pmr::memory_resource>
+            cUpstreamResource{&upstream_buffer, SmallMessageSizeBytes, cetl::pmr::null_memory_resource()};
+        cetl::pmr::UnsynchronizedArrayMemoryResource<cetl::pmr::memory_resource> cResource{small_message_buffer_,
+                                                                                           SmallMessageSizeBytes,
+                                                                                           &cUpstreamResource};
+        cetl::pmr::polymorphic_allocator<std::uint64_t>                          cAlloc{&cResource};
+        Message                                                                  c{cAlloc};
+
+        // This time we won't reserve which should cause vector to do multiple allocations. We'll also insert
```

Comment:
Fix comment.  "We'll also insert a bunch more" is not true.

Reply:
Good catch. Did a bunch of cleanup of the example text here.